File: notification-service/pkg/logger/logger.go
package logger

import (
	"log"
	"os"
)

var logger *log.Logger

func init() {
	logger = log.New(os.Stdout, "[NOTIFICATION_SERVICE] ", log.LstdFlags|log.Lshortfile)
}

func Info(msg string) {
	logger.Println("INFO:", msg)
}

func Error(msg string) {
	logger.Println("ERROR:", msg)
}

func Debug(msg string) {
	logger.Println("DEBUG:", msg)
}

func Warn(msg string) {
	logger.Println("WARN:", msg)
}

File: notification-ai-service/app/utils/logger.py
import logging
import sys

# Configure logger
logger = logging.getLogger('notification_ai_service')
logger.setLevel(logging.DEBUG)

# Create console handler
handler = logging.StreamHandler(sys.stdout)
handler.setLevel(logging.DEBUG)

# Create formatter
formatter = logging.Formatter('[NOTIFICATION_AI_SERVICE] %(asctime)s - %(levelname)s - %(message)s')
handler.setFormatter(formatter)

# Add handler to logger
if not logger.handlers:
    logger.addHandler(handler)

def info(msg: str):
    logger.info(msg)

def error(msg: str):
    logger.error(msg)

def debug(msg: str):
    logger.debug(msg)

def warning(msg: str):
    logger.warning(msg)